R/spatial_autocorrelation.R

```diff
@@ -197,9 +201,19 @@ CorSpatialFeatures.default <- function (
 
     if (verbose) cli_alert("  Computed feature spatial autocorrelation scores")
 
+    if (calculate_pvalue) {
+      pvals <- .corPValues(cor_values = spatial_autocorrelation, x = x_subset, ...)
+      if (verbose) cli_alert("  Computed p-values for spatial autocorrelation scores")
+    }
+    
     # Summarize results
-    results <- tibble(gene = names(spatial_autocorrelation), cor = spatial_autocorrelation) |>
-      arrange(-cor)
+    results <- tibble(gene = names(spatial_autocorrelation), cor = spatial_autocorrelation)
+    if (calculate_pvalue) {
+      results <- results |> 
+        add_column(pval = pvals) |> 
+        add_column(adj.pval = p.adjust(pvals, method = "BH"))
```

Comment:
Perhaps there should be an option to set the MH adjustment method?

Reply:
I thought about that as well, but it's such an easy thing to do afterwards yourself so I might leave it like this (for now at least).